llvm/lib/Transforms/InstCombine/InstCombineAndOrXor.cpp

```diff
@@ -2708,6 +2752,11 @@ Instruction *InstCombinerImpl::visitAnd(BinaryOperator &I) {
   if (Instruction *Res = foldBitwiseLogicWithIntrinsics(I, Builder))
     return Res;
 
+  if (Value *V = simplifyAndOrWithOpReplaced(Op0, Op1, /*IsAnd*/ true, *this))
+    return BinaryOperator::CreateAnd(Op1, V);
```

Comment:
```suggestion
    return BinaryOperator::CreateAnd(V, Op1);
```
No semantic difference, but I'd try to preserve operand order in cases where it's easy to do, like here.